packages/lambda-xyz/src/tile.set.cache.ts

```diff
@@ -9,20 +9,104 @@ export function getTileSet(name: string, projection: Epsg): TileSet | undefined
     return TileSets.get(tileSetId);
 }
 
-export async function loadTileSet(req: LambdaContext, name: string, projection: Epsg): Promise<TileSet | null> {
-    req.set('tileSet', name);
-    req.set('projection', projection);
+export async function loadTileSet(name: string, projection: Epsg): Promise<TileSet | null> {
     let tileSet = getTileSet(name, projection);
     if (tileSet == null) {
         tileSet = new TileSet(name, projection);
         TileSets.set(tileSet.id, tileSet);
     }
-    req.timer.start('tileset:load');
     const loaded = await tileSet.load();
-    req.timer.end('tileset:load');
     if (!loaded) {
         TileSets.delete(tileSet.id);
         return null;
     }
     return tileSet;
 }
+
+function titleizeName(name: string): string {
+    return name.replace(/_0-/g, ' 0.').replace(/_/g, ' ');
+}
+
+function individualTileSet(parent: TileSet, image: TileSetRuleImagery): TileSet {
+    const id = image.imagery.id.slice(3);
+    const copy = new TileSet(id, parent.projection);
+    // use parent data as prototype for child;
+    copy.tileSet = Object.create(parent.tileSet ?? null);
+
+    const { tileSet } = copy;
+
+    copy.titleOverride = `${parent.title} ${titleizeName(image.imagery.name)}`;
+    copy.extentOverride = Bounds.fromJson(image.imagery.bounds).toBbox();
+
+    const rule = {
+        id,
+        minZoom: 0,
+        maxZoom: image.rule.maxZoom,
+        priority: 0,
+    };
+    // Override imagery (for completeness)
+    tileSet.imagery = { [id]: rule };
+    copy.tileSet = tileSet;
+
+    copy.imagery = [
+        {
+            rule,
+            imagery: image.imagery,
+        },
+    ];
+
+    return copy;
+}
+
+async function collectTileSets(tileSets: TileSet[], name: string, projection: Epsg | null): Promise<void> {
```

Comment:
ideally we should not be modifying stuff we pass into functions, is there a reason we need to "collect" rather than just return the list of tile sets?

eg 
```
getTileSets(name: string, projection?: Epsg): Promise<TileSet[]>
```

Reply:
also do we ever dedupe this?

Reply:
When it is a private function then I see no problem with collecting the in the list; the name says what it's doing.

Reply:
dedupe is not required here because a tileSet does not contain duplicates.